src/Language/PureScript/TypeChecker/TypeSearch.hs

```diff
@@ -85,6 +110,9 @@ typeSearch
   -- ^ The typechecker state
   -> P.Type
   -- ^ The type we are looking for
-  -> Map (P.Qualified P.Ident) P.Type
+  -> ([(P.Qualified P.Ident, P.Type)], Maybe [(Label, Type)])
```

Comment:
I think there is a misunderstanding here:

`?x {x: 1, y: "hello"} :: Int` will find both `_.x` as well as `_.y`, so I don't think we should mix these results with the others, because they don't make the code type check necessarily. It might be a nice idea to actually add another check and add *just*  `_.x` to the first list.